src/paper/views/paper_views.py

```diff
@@ -583,9 +582,21 @@ def eligible_reward_summary(self, request, pk=None):
         """
         Provides information about rewards for which this paper is eligible for
         """
+        paper_id = pk
+        if not paper_id:
+            return Response("paper_id is required", status=400)
+
+        paper = Paper.objects.get(id=paper_id)
+        if not paper:
+            return Response("Paper not found", status=404)
+
+        try:
+            paper_rewards = paper.paper_rewards
+        except Exception:
+            return Response("Failed to get paper reward", status=400)
```

Comment:
As we are past argument validation, we should return a server error (500 series).